File: backend/src/main/java/de/neuefische/capstone/backend/model/Scorecard.java
package de.neuefische.capstone.backend.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Data
@With
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Document("Scorecards")
public class Scorecard {
    @Id
    private String scorecardId;
    private String userId;
    private String golfCourseId;
    private String golfCourseName;
    private String golfClubName;
    private List<String> players;
    private String date;
    private List<Score> scores;
    private int totalScore;
    private boolean playBackNine;
    private double CourseRating;
    private int SlopeRating;
    private double CourseHandicap;
}

File: frontend/src/models/Scorecard.ts
export type Scorecard = {
    scorecardId: string;
    userId: string;
    golfCourseId: string;
    golfCourseName: string;
    golfClubName: string;
    players: string[];
    date: string;
    scores: Score[];
    totalScore: number;
    playBackNine: boolean;
    courseRating: number;
    slopeRating: number;
    courseHandicap: number;
    coursePar: number;
}

export type Score = {
    holeNumber: number;
    totalStrokes: number;
    totalPutts: number;
    fairwayHit: boolean;
}

export type ScorecardDTO = {
    userId: string;
    golfCourseId: string;
    golfCourseName: string;
    golfClubName: string;
    players: string[];
    date: string;
    scores: Score[];
    totalScore: number;
    playBackNine: boolean;
    courseRating: number;
    slopeRating: number;
    courseHandicap: number;
    coursePar: number;
}